csharp/autobuilder/Semmle.Autobuild.Shared/BuildCommandAutoRule.cs

```diff
@@ -10,23 +10,51 @@ namespace Semmle.Autobuild.Shared
     public class BuildCommandAutoRule : IBuildRule<AutobuildOptionsShared>
     {
         private readonly WithDotNet<AutobuildOptionsShared> withDotNet;
+        private IEnumerable<string> candidatePaths;
+        private string? scriptPath;
+
+        /// <summary>
+        /// A list of paths to files in the project directory which we classified as scripts.
+        /// </summary>
+        public IEnumerable<string> CandidatePaths
+        {
+            get { return this.candidatePaths; }
+        }
+
+        /// <summary>
+        /// The path of the script we decided to run, if any.
+        /// </summary>
+        public string? ScriptPath
```

Comment:
Is it possible to collapse into an auto property with public get and private set?